dataprofiler/profilers/utils.py

```diff
@@ -567,6 +569,8 @@ def find_diff_of_dicts_with_diff_keys(dict1, dict2):
                 diff_shared[key] = find_diff_of_dates(value1, value2)
             elif isinstance(value1, str) or isinstance(value1, bool):
                 diff_shared[key] = find_diff_of_strings_and_bools(value1, value2)
+            elif isinstance(value1, dict) and isinstance(value2, dict):
```

Comment:
need to update `test_utils.py` as well for this change

Reply:
tests have now been added to test the nested dict functionality